qiskit/algorithms/amplitude_amplifiers/amplification_problem.py

```diff
@@ -63,12 +62,12 @@ def __init__(
         if is_good_state is not None:
             self._is_good_state = is_good_state
         elif hasattr(oracle, "evaluate_bitstring"):
-            self._is_good_state = oracle.evaluate_bitstring
+            self._is_good_state = oracle.evaluate_bitstring  # type: ignore[union-attr]
```

Comment:
This ignore is debatable. The `oracle` of either `QuantumCircuit`/`StateVector` might have a properly typed `evaluate_bitstring`. I think you can remove this line change and leave the error to be fixed by a future PR.

Reply:
Oh, I didn't see the `hasattr(oracle, "evaluate_bitstring")` constraint. In that case, this line is good. Just ignore my comment.